app/services/landingPageTourService.ts

```diff
@@ -0,0 +1,120 @@
+import * as angular from 'angular';
+import * as jQuery from 'jquery';
+
+interface ILandingPageTourService {
+  startTour(tourConfig: any);
+}
+
+export class LandingPageTourService implements ILandingPageTourService {
+  public static $inject = ['$document', '$timeout', 'Constants', 'GuidedTourService'];
+
+  private $document: any;
+  private $timeout: any;
+  private Constants: any;
+  private GuidedTourService: any;
+  private tourConfig: any;
+
+  constructor ($document: any, $timeout: any, Constants: any, GuidedTourService: any) {
+    this.$document = $document;
+    this.$timeout = $timeout;
+    this.Constants = Constants;
+    this.GuidedTourService = GuidedTourService;
+
+    this.tourConfig = {
+      id: "getting-started-tour",
+      steps: []
+    };
+
+    let stepConstants = Constants.LANDING_PAGE_TOUR_STEPS || {};
```

Comment:
Yes, an inline function in the tour JSON probably. We already support something like it for left nav with `isValid`.

https://docs.openshift.org/latest/install_config/web_console_customization.html#project-left-navigation

> They would have to be intimate with the page structure and/or class names for that as well.

If they want to completely replace the tour, this is true. I think the biggest problem is that it could break on upgrade if we change things.